src/routes/ttn.js

```diff
@@ -0,0 +1,93 @@
+// sse listeners
+// {'001a7988170187ca': [{requestId: {request, reply, messageId}}]}
+const subscribersMap = new Map();
+
+// only binary data in hex
+// {'001a7988170187ca': '2003'}
+const payloads = new Map();
+
+
+export default async fastify => {
+    fastify.post(
+        '/webhook',
+        async request => {
+            const {body} = request;
+            const {device_id: id} = body.end_device_ids;
+            const subscribers = subscribersMap.get(id);
+
+            if ( subscribers ) {
+                let payload = body?.downlink_message?.frm_payload;
+
+                // extend downlink message from buffer
+                if ( payload ) {
+                    payload = payloads.get(id)?.pop();
+                    // should be used only once and then cleared
+                    payloads.delete(id);
+                } else {
+                    payload = body?.uplink_message?.frm_payload;
+                }
+
+                // convert to hex
+                if ( payload ) {
+                    payload = Buffer.from(payload, 'base64').toString('hex');
+                }
+
+                for ( const subscriber of subscribers.values() ) {
+                    subscriber.reply.sse({
+                        id: subscriber.messageId++,
+                        data: JSON.stringify({
+                            id,
+                            time: new Date(body.received_at).getTime(),
+                            data: payload,
+                            info: body
+                        })
+                    });
+                    console.log('send sse to', subscriber.messageId);
+                }
+            }
+        }
+    );
+
+    fastify.post(
+        '/:id/messages',
```

Comment:
```suggestion
        '/:eui/messages',
```
?

or change to id in chirpstack

Reply:
changed to `id` in chirpstack